src/auth/auth.controller.ts

```diff
@@ -31,4 +31,41 @@ export class AuthController {
     async register(@Body() createUserDto: CreateUserDto): Promise<SanitizedUser> {
         return await this.authService.register(createUserDto);
     }
+
+    // POST /auth/confirm/:token
+    // This path lets the user confirm his email
+    @Public()
+    @Post('confirm')
+    async confirmEmail(@Query('token') token: string): Promise<string> {
+        if (!await this.authService.confirmEmail(token)) {
+            throw new HttpException("No user with these specifications has been found.", HttpStatus.BAD_REQUEST)
+        }
+        return "Email successfully verified";
+
+    }
+
+
+    // POST /auth/password/request
+    // This path lets the user request to change his password
+    @Public()
+    @Post('password/request')
+    async requestPassword(@Body() passwordRequestDto: PasswordRequestDto): Promise<string> {
+        await this.authService.passwordRequest(passwordRequestDto)
+        // This function will always return this message for security reasons
+        return "If there was an account with this email, they should receive an email with the instructions on how to change their password.";
+
+    }
+
+
+    // POST /auth/password/reset
+    // This path lets the user reset his password
+    @Public()
+    @Post('password/reset')
+    async resetPassword(@Query('token') token: string, @Body() passwordResetDto: PasswordResetDto): Promise<string> {
```

Comment:
Using `@Param()` is preferable to `@Query()` when possible. It uses the request parameters instead of looking at the query.

```suggestion
    @Post('password/reset/:token')
    async resetPassword(@Param('token') token: string, @Body() passwordResetDto: PasswordResetDto): Promise<string> {
```